packages/grafana-ui/src/components/Table/TableCell.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { FC } from 'react';
+import { Cell, CellProps, TableCellProps } from 'react-table';
+import { Field } from '@grafana/data';
+import { BackgroundColoredCell, DefaultCell } from './DefaultCell';
+import { BarGaugeCell } from './BarGaugeCell';
+import { getTextAlign } from './utils';
+import { TableCellDisplayMode, TableFieldOptions, TableFilterActionCallback } from './types';
+import { TableStyles } from './styles';
+
+interface Props {
+  cell: Cell;
+  field: Field;
+  tableStyles: TableStyles;
+  onCellClick?: TableFilterActionCallback;
+}
+
+export const TableCell: FC<Props> = ({ cell, field, tableStyles, onCellClick }) => {
+  const filterable = field.config.filterable;
+  const cellProps = cell.getCellProps();
+
+  let onClick: ((event: React.SyntheticEvent) => void) | undefined = undefined;
+  if (filterable && onCellClick) {
+    if (cellProps.style) {
+      cellProps.style.cursor = 'pointer';
+    }
+
+    onClick = () => onCellClick(cell.column.Header as string, cell.value);
+  }
+  const fieldTextAlign = getTextAlign(field);
+  if (fieldTextAlign && cellProps.style) {
+    cellProps.style.textAlign = fieldTextAlign;
+  }
+
+  cell.column.Cell = getCellDisplayType(cellProps, field, tableStyles);
+
+  return (
+    <div {...cellProps} onClick={onClick}>
+      {cell.render('Cell')}
+    </div>
```

Comment:
need to move:
cell.column.Cell = getCellDisplayType(cellProps, field, tableStyles);
to getColumns (where it was before).  

and pass field & tableStyles as extra pros in the cell.render call